control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go

```diff
@@ -2131,6 +2143,22 @@ func (r *reconciler) ensureServiceLoadBalancersRemoved(ctx context.Context) (boo
 	return removed, nil
 }
 
+func (r *reconciler) ensureVolumeSnapshotsRemoved(ctx context.Context) (bool, error) {
+	log := ctrl.LoggerFrom(ctx)
+	vss := &snapshotv1.VolumeSnapshotList{}
+	if err := r.client.List(ctx, vss); err != nil {
```

Comment:
You will need to add a `client.ListOptions` if you implement the suggestion from the comment above.